src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -501,17 +512,41 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
         const node = elements.find((element) => element.id === selectedModelId);
         if (node) {
             currentNodeIdRef.current = node.id;
-            const modelClicked = {
-                '@id': node.id,
-                '@type': node.data.type,
-                '@context': node.data.context,
-                displayName: node.data.name,
-                contents: node.data.content
-            };
+            let modelClicked = null;
+            if (node.type === OATInterfaceType) {
+                modelClicked = {
+                    '@id': node.id,
+                    '@type': node.data.type,
+                    '@context': node.data.context,
+                    displayName: node.data.name,
+                    contents: node.data.content
+                };
+            } else {
+                modelClicked = new DTDLRelationship(
+                    node.id,
+                    node.data.name,
+                    node.data.displayName,
+                    node.data.description,
+                    node.data.comment,
+                    node.data.writable,
+                    node.data.content ? node.data.content : [],
+                    node.data.target,
+                    node.data.maxMultiplicity
+                );
+
+                if (node.data.type === OATExtendHandleName) {
+                    modelClicked['@type'] = OATExtendHandleName;
+                }
+            }
             dispatch({
                 type: SET_OAT_PROPERTY_EDITOR_MODEL,
                 payload: modelClicked
             });
+        } else {
+            dispatch({
+                type: SET_OAT_PROPERTY_EDITOR_MODEL,
+                payload: null
+            });
```

Comment:
Maybe we pull this out of the if/else block and have it run at the end of the method with the value of `modelClicked`? And have the default value of that be null?